cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/constraint/CheckableConstraint.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.schedule.constraint;
+
+import co.cask.cdap.internal.app.runtime.schedule.ProgramSchedule;
+
+/**
+ * A constraint that can be checked for satisfaction.
+ */
+public interface CheckableConstraint {
```

Comment:
hum.. now this becomes interesting, since this interface doesn't not extend from `Constraint` (which it can't).